### PR TITLE
 Laboratory work #2, Baranov Timofey - 22FPL1

### PR DESCRIPTION
**Pull request by tima-baranov**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -14,23 +13,24 @@ def tokenize(text: str) -> list[str] | None:
     """
     if not isinstance(text, str):
         return None
-
-    return [token.lower() for token in text if token.isalpha()]
-
-
+    tokens = text.lower()
+    for token in tokens:
+        if token in punctuation:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:18]: looks like we can use single `if`

File: lab_2_tokenize_by_bpe/main.py [hunk 2]
```diff
@@ -14,8 +14,15 @@ def prepare_word(
...
-
+    if not isinstance(raw_word, str) or not isinstance(start_of_word, str | None) \
+            or not isinstance(end_of_word, str | None):
+        return None
+    tokens = list(raw_word)
+    if start_of_word is not None:
+        tokens.insert(0, start_of_word)
+    if end_of_word is not None:
+        tokens.append(end_of_word)
+    return tuple(tokens)
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:25]: Can you add some empty lines?

File: lab_2_tokenize_by_bpe/main.py [hunk 3]
```diff
@@ -26,8 +33,15 @@ def collect_frequencies(
...
-
+    if not isinstance(text, str) or not isinstance(start_of_word, str | None) or not isinstance(end_of_word, str):
+        return None
+    freq_dictionary = {}
+    for i in text.split():
+        preprocessed_word = prepare_word(i, start_of_word, end_of_word)
+        if not preprocessed_word:
+            return None
+        freq_dictionary.update({preprocessed_word: text.split().count(i)})
+    return freq_dictionary
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:44]: the same

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_tokenize_by_bpe/main.py (hunk 2 above)
Comment (tima-baranov) [reply lab_2_tokenize_by_bpe/main.py:25]: Fixed

File: lab_2_tokenize_by_bpe/main.py (hunk 3 above)
Comment (tima-baranov) [reply lab_2_tokenize_by_bpe/main.py:46]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -2,7 +2,6 @@
 Lab 1
 Language detection
 """
-import json
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:5]: Revert, please

File: lab_2_tokenize_by_bpe/main.py [hunk 5]
```diff
@@ -38,6 +56,11 @@ def count_tokens_pairs(
     """
 
 
+
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:59]: rm

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (tima-baranov) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_2_tokenize_by_bpe/main.py (hunk 5 above)
Comment (tima-baranov) [reply lab_2_tokenize_by_bpe/main.py]: Fixed

File: lab_2_tokenize_by_bpe/main.py [hunk 6]
```diff
@@ -26,6 +35,15 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(text, str) or not isinstance(start_of_word, str | None) or not isinstance(end_of_word, str):
+        return None
+    freq_dictionary = {}
+    for i in text.split():
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:41]: let's rename `i`

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_tokenize_by_bpe/main.py (hunk 6 above)
Comment (tima-baranov) [reply lab_2_tokenize_by_bpe/main.py]: Fixed

**Pull request closed without merging**